tokio/src/runtime/task/core.rs

```diff
@@ -157,8 +157,8 @@ pub(crate) struct Header {
     /// Table of function pointers for executing actions on the task.
     pub(super) vtable: &'static Vtable,
 
-    /// This integer contains the id of the OwnedTasks or LocalOwnedTasks that
-    /// this task is stored in. If the task is not in any list, should be the
+    /// The lower 32 bits of this value is the id of OwnedTasks, and higher 32 bits of this value is
+    /// the segment id of inner segment list of OwnedTasks. If the task is not in any list, should be the
```

Comment:
Why not just recompute this every time you access the list? I think that would simplify the code.

Is the problem with being able to access the task id inside `list.rs`?

Reply:
It is because if we use task_id to calculate the index every time, we have to store index in Header struct. But Core struct also relies on task_id to generate panic and some logic in TaskIdGuard, which means we either store a duplicate copy of task_id in Core(just like header), or pass in task_id as a parameter when calling Core-related methods.

The reason for this is not to avoid calculating the index every time, but so that thead Header struct does not need to know the task_id. 

Header knows the index and OwnedTasks, Core konws the scheduler and task_id. What do you think of it?

Reply:
Depend on the current design, when we got a task in `list.rs`, we can access its Header, and have no ideas about Core, the Core is used by the scheduler.

Reply:
It definitely works! So I will use this method to caculate the index.

Reply:
I have use `Header::get_id` method to calculate the index, and it works well.